python/cugraph/cugraph/structure/property_graph.py

```diff
@@ -1748,6 +1928,180 @@ def renumber_edges_by_type(self):
+    ):
+        """Check if vector_properties is valid and update vector_property_lengths"""
+        df_cols = set(df.columns)
+        for key, columns in vector_properties.items():
+            if key in invalid_keys:
+                raise ValueError(
+                    "Cannot assign new vector property to existing "
+                    f"non-vector property: {key}"
+                )
+            if isinstance(columns, str):
+                # If df[columns] is a ListDtype column, should we allow it?
+                raise TypeError(
+                    f"vector property columns for {key!r} should be a list; "
+                    f"got a str ({columns!r})"
+                )
+            if not df_cols.issuperset(columns):
+                missing = ", ".join(set(columns) - df_cols)
+                raise ValueError(
+                    f"Dataframe does not have columns for vector property {key!r}:"
+                    f"{missing}"
+                )
+            if not columns:
+                raise ValueError("Empty vector property columns for {key!r}!")
+            if vector_property_lengths.get(key, len(columns)) != len(columns):
+                prev_length = vector_property_lengths[key]
+                new_length = len(columns)
+                raise ValueError(
+                    f"Wrong size for vector property {key}; got {new_length}, but "
+                    f"this vector property already exists with size {prev_length}"
+                )
+        for key, columns in vector_properties.items():
+            vector_property_lengths[key] = len(columns)
+
+    def _create_vector_properties(self, df, vector_properties):
+        if self.__series_type is cudf.Series:
+            ascontiguousarray = cupy.ascontiguousarray
+        else:
+            ascontiguousarray = np.ascontiguousarray
+        # Make each vector contigous and 1-d
+        vectors = {
+            key: [
+                np.squeeze(vec, 0)
+                for vec in np.split(ascontiguousarray(df[columns].values), len(df))
+            ]
+            for key, columns in vector_properties.items()
+        }
+        for key, vec in vectors.items():
+            df[key] = vec
+
+    def _get_vector_property(self, df, col_name, length, fillvalue, missing):
+        if type(df) is not self.__dataframe_type:
+            raise TypeError(
+                f"Expected type {self.__dataframe_type}; got type {type(df)}"
+            )
+        if col_name not in df.columns:
+            raise ValueError(f"Column name {col_name} is not in the columns of df")
+        if missing not in {"error", "ignore"}:
+            raise ValueError(
+                f'missing keyword must be one of "error" or "ignore"; got {missing!r}'
+            )
+        if fillvalue is not None:
+            try:
+                fill = list(fillvalue)
+            except Exception:
+                fill = [fillvalue] * length
```

Comment:
Can we not use  list here but numpy or cupy/array library here please

Reply:
I think we need to use list here; it's the only way I could get it to work.

Reply:
Gotcha, I think that's fine for now, we can revisit this when we start perf tuning if this becomes a problem